src/compiler/core.ts

```diff
@@ -515,12 +525,23 @@ namespace ts {
         return result || array;
     }
 
+    export function mapAllOrFail<T, U>(array: ReadonlyArray<T>, mapFn: (x: T, i: number) => U | undefined): U[] | undefined {
+        const result: U[] = [];
+        for (let i = 0; i < array.length; i++) {
+            const mapped = mapFn(array[i], i);
+            if (!mapped) {

```

Comment:
Should this be an explicit check for `undefined`?  Or is a falsy value considered to indicate failure?